File: 백준/Bronze/11721. 열 개씩 끊어 출력하기/열 개씩 끊어 출력하기.cc
#include <iostream>

using namespace std;

int main() {
    string str;

    cin >> str;

    int currentStrIndex = 0;
    int tenUnitCounter = 0;
    while (true)
    {
        for (int i = 0; i < 10; i++) {
            cout << str[tenUnitCounter + i];
            currentStrIndex++;
            if (currentStrIndex == str.size()) {
                break;
            }
        }

        if (currentStrIndex == str.size()) {
            break;
        }

        tenUnitCounter = tenUnitCounter + 10;
        cout << "\n";
    }

    return 0;
}

File: 백준/Bronze/11721. 열 개씩 끊어 출력하기/열 개씩 끊어 출력하기.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String str;

        str = sc.nextLine();

        int currentStrIndex = 0;
        int tenUnitCounter = 0;
        while (true) {
            for (int i = 0; i < 10; i++) {
                System.out.print(str.charAt(tenUnitCounter + i));
                currentStrIndex++;
                if (currentStrIndex == str.length()) {
                    break;
                }
            }

            if (currentStrIndex == str.length()) {
                break;
            }

            tenUnitCounter = tenUnitCounter + 10;
            System.out.print("\n");
        }

        sc.close();
    }
}
